modules/social_features/social_group/src/Element/SocialGroupEntityAutocomplete.php

```diff
@@ -61,9 +61,30 @@ public static function validateEntityAutocomplete(array &$element, FormStateInte
         ];
 
         $account = User::load($match);
```

Comment:
I think it makes sense to change this to

```suggestion
        $account = User::load($match);
        if ($account === NULL) {
          throw new \RuntimeException("Input matched a user that was supposed to exist but couldn't be loaded. This indicates an implementation error where we may be mixing up entity types.");
        }
```

That avoids your individual `$account !== NULL` changes.  

The functionality in this function relies on vaildating some kind of user, so if we don't have that user we can't continue.

Additionally in your current imlpementation we have `if ($account !== NULL && !$user_has_role) {` as potential error. In this case if we don't have a user we know for sure we don't have the role which would allow them to be added. So I would expect some kind of error to come up. However, because we have no user, we're not providing an error. That indicates an issue with our implementation that throwing a clear error makes explicit.

Reply:
Thanks, I added this change